htlcswitch/interceptable_switch.go

```diff
@@ -416,6 +445,60 @@ func (s *InterceptableSwitch) resolve(res *FwdResolution) error {
 	}
 }
 
+func (s *InterceptableSwitch) validateChannelUpdate(
+	update *lnwire.ChannelUpdate) error {
+
+	// The maxHTLC flag is mandatory.
+	if !update.MessageFlags.HasMaxHtlc() {
```

Comment:
More validation is needed here, eg: max HTLC shouldn't be greater than the channel capacity, etc, etc. I think we should just call the normal validation routine.

Reply:
Wanted to call

https://github.com/lightningnetwork/lnd/blob/bf5aab9d5246daede3f499af880a5d7be6c6ee88/routing/ann_validation.go#L165

But that would create a circular reference. For now, I just copied over the `max_htlc >= min_htlc` check. Capacity checking is not possible, because the real lnd doesn't know what the capacity is. It's probably infinite because the channel is typically virtual.

If desired I can do a bigger refactor and move the validation logic into `htlcswitch`. But not sure if worth it.